_drafts/cortexm-and-clang.md

```diff
@@ -0,0 +1,971 @@
+---
+title: "Taking advantage of LLVM / Clang in Embedded Projects"
+description: "An overview of how to compile an ARM Cortex-M based project with LLVM/Clang and an exploration of the neat features available in the toolchain such as static analysis to identify memory leaks and deadlocks"
+image:
+tag: [cortex-m]
+author: chris
+---
+
+If you have ever done any software development that involves Apple products, FreeBSD, WebAssembly, or Rust you have made use of a wildly popular compiler toolchain known as LLVM. However, in the embedded world, LLVM and the open source C-language compiler built on top of it (Clang) still do not get a lot of attention.
+
+<!-- excerpt start -->
+
+In this article, I hope I can convince you that adding a Clang build target to your project can be
+a relatively simple and useful endeavor. We will cover how to take advantage of some of the
+rich features shipped with the toolchain and identify some memory leaks and mutex deadlocks in an
+example project at compile time! We will also walk through a step-by-step example of updating a GCC
+based ARM Cortex-M project to compile you need to take to cross-compile with LLVM/Clang.
+
+<!-- excerpt end -->
+
+> Note: While the focus of the article will be using LLVM/Clang with ARM Cortex-M embedded devices, the
+> general overview of LLVM as well as the features leveraged can be applied to any architecture
+> supported by the toolchain.
+
+{:.no_toc}
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+* auto-gen TOC:
+{:toc}
+
```

Comment:
I think you do an OK job at mentioning the benefits of compiling with Clang, but I'm missing a section that makes it trivial for me to understand why I would benefit from compiling my project with clang or any other compiler. 


## Benefits of Compiling with Clang

Many embedded projects use IAR, armcc, GCC, or one of other proprietary build systems. blah blah

- Static analysis
- Easier to move off of a proprietary compiler when the time is right by becoming less dependent on a single compiler feature set
- You can use both compilers `-Werror` flag to your advantage and reap the most amount of benefit from compile time analysis
- By compiling with clang, you are setting yourself up nicely for running unit tests and simulated builds on host



I think the section probably belongs after the background section